### PR TITLE
[#1999] Provider search links to the provider page

### LINKED ISSUE #1999: result in main searchbar should only show a matched provider in provider section
Tested on v3.10.0-beta.1
**Describe the bug**
after entering the word in main searchbar, the result in provider section should be the name of the provider without any services assigned

**To Reproduce**
1. Go to '/services'
2. Enter in main searchbar "EGI"

**Expected behavior**
in section Provider should only be visible name of matched provider,  here "EGI Foundation". When the user click on the provider name shoulbe be redirected to provider page. 

**Screenshots**
image

### PR DESCRIPTION
**Pull request by goreck888**
Change link in autocomplete for providers
to the provider_path
Fixes #1999

### CODE REVIEW

**Review by jswk — CHANGES_REQUESTED** [review]
There is a regression regarding service authorization.
It is now possible to find deleted or draft services in all the search bars, not only in backoffice.
image
Clicking it leads to:
image

**Comment by Marcelinna** [comment]
Master
image

PR
image

**Comment by goreck888** [comment]
> There is a regression regarding service authorization.
> It is now possible to find deleted or draft services in all the search bars, not only in backoffice.
> image
> Clicking it leads to:
> image

Fixed.

**Comment by goreck888** [comment]
> Master
> image
> 
> PR it is problem with services
> image

We don't have too much influence on what searchkick returns. I try to change the limit of results to 10, it should be better.

File: spec/features/search_spec.rb [hunk 1]
```diff
@@ -40,19 +40,45 @@
...
+  end
+
+  scenario "redirect to provider path when selecting provider_id by autocomplete controller", js: true, search: true do
+    provider = create(:provider)
+    fill_in "q", with: provider.name
+    expect(page).to have_css("#-option-0")
+    find(:css, "li[id='-option-0']").click
+    expect(page).to have_content(provider.name)
+    expect(current_path).to eq(provider_path(provider))
+  end
```
Comment (jswk) [inline spec/features/search_spec.rb:68]: Something is off with this test failing. I cannot pinpoint a concrete place, but when I run the test without headless option for webdriver, it's impossible to find the created provider. Either the provider isn't synced to ES or the search query in autocomplete.rb is done incorrectly?
What's strange is that providers appear in development, or on the testing docker instance, but not in this test.

**Review by jswk — CHANGES_REQUESTED** [review]
These two test issues are to be addressed, otherwise only the changelog is missing.

File: spec/features/search_spec.rb (hunk 1 above)
Comment (goreck888) [reply spec/features/search_spec.rb]: Resolved. The problem was the created provider wasn't reindexed. Now this test pass.

**Review by jswk — CHANGES_REQUESTED** [review]
Aaand, changelog!

**Review by jswk — APPROVED** [review]

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**